netket/graph/continuous_geometry.py

```diff
@@ -0,0 +1,146 @@
+            dim: (Optional) The number of spatial dimensions in the Hilbert space
+            lattice: (Optional) The lattice that describes the Hilbert space. For 'pbc=True' this is the basis of the
+                        physical simulation cell. For 'pbc=False' the standard Euclidean basis is assumed.
+            geometry: (Optional) A geometry object. Either a periodic box 'Cell' or free space 'Free'.
+        """
+        if dim is None and lattice is None:
+            if isinstance(pbc, bool):
+                raise ValueError(
+                    """If 'dim' and 'lattice' is not given, the 'pbc' has to be a tuple to infer the spatial
+                    dimensions"""
+                )
+            dim = len(pbc)
+
+        elif lattice is None:
+            self._dim = dim
+        else:
+            self._dim = jnp.shape(lattice)[0]
+
+        if isinstance(pbc, bool):
+            self._pbc = (pbc,) * self._dim
+
+        if lattice is None:
+            # assume cubic box if lattice is not given
+            self._lat = jnp.eye(self._dim)
+        else:
+            self._lat = jnp.array(lattice)
+
+        self._ilat = jnp.linalg.inv(self._lat)
+
+        if not self._dim == len(self._pbc) == jnp.shape(self._lat)[0]:
+            raise ValueError(
+                """There is a mismatch between the number of spatial dimensions
+                inferred by 'pbc', 'lattice' and 'dim."""
+            )
+        self._hash = None
+
+    @property
+    def pbc(self) -> Tuple:
+        r"""Whether or not periodic boundary conditions are applied along a spatial dimension (x,y,z,...)"""
+        return self._pbc
+
+    @property
+    def dim(self):
+        r"""The spatial dimensions of the Hilbert space"""
+        return self._dim
+
+    @property
+    def lattice(self):
+        r"""The lattice used."""
+        return self._lat
+
+    @property
+    def inv_lattice(self):
+        r"""The reciprocal lattice to the lattice."""
+        return self._ilat
+
+    @property
+    def volume(self) -> int:
+        r"""The volume of the Hilbert space if defined (only defined if it is fully periodic)"""
+        raise NotImplementedError
+
+    @property
+    def extent(self):
+        r"""Spatial extension in each spatial dimension"""
+        raise NotImplementedError
+
+    def distance(self, x, y=None, norm=False, mode=None) -> list[list]:
+        """x: (..., N, dim)
+        y: None or (..., N, dim)
+        norm: whether or not the norm of distances is returned as well
+        mode: mode of distance computation
+        """
+        raise NotImplementedError
+
+    def back_to_box(self, x):
+        r"""Given coordinates in space, this function puts the coordinates back to the physical space.
+        For free space this is the identity operation. For periodic boxes the fractional coordinates are
+        put between 0 and 1."""
+        raise NotImplementedError
+
+    def from_lat_to_standard(self, x):
```

Comment:
yes this should not be in the base class it's a specific thing of `Cell` I guess

Reply:
I think it is a naming issue. What lattice is supposed to be is a basis of the physical space. Standard is the standard basis e.g. (1.0),(0,1) in 2D. So a better naming instead of `lattice` is probably `basis`. The two functions then allow to transform back and forth between the two bases.
I actually think it belongs to the base class. I‘m not sure of a use case right now but also in free space it might make sense to define another basis than the standard basis.